internal/repository/userRepository.go

```diff
@@ -29,48 +30,56 @@ func NewUserRepository(client *mongo.Client, dbName string, collectionName strin
 		collection: collection,
 	}
 }
-func (repo *repository) Save(ctx context.Context, user models.User) (models.User, error) {
+func (repo *repository) Save(ctx context.Context, user models.User) (*models.User, error) {
 	logger.Infof("Saving user %s", user.GetFirstName())
 	newUser := mapper.UserToUserEntity(user)
 	_, err := repo.collection.InsertOne(ctx, newUser)
 	if err != nil {
 		logger.Errorf("Error saving user %s: %v", user.GetFirstName(), err)
-		return models.User{}, err
+		return nil, err
 	}
 	logger.Infof("User %s saved successfully", user.GetFirstName())
 	createdUser, err := mapper.UserEntityToUser(newUser)
-	return createdUser, nil
+	return &createdUser, nil
 }
 
-func (repo *repository) GetUserById(ctx context.Context, id string) (models.User, error) {
+func (repo *repository) GetUserById(ctx context.Context, id string) (*models.User, error) {
 	logger.Infof("Getting user by id %s", id)
 	var userEntity entities.UserEntity
 
 	uid, err := uuid.Parse(id)
 	if err != nil {
 		logger.Errorf("Error parsing user id %s: %v", id, err)
-		return models.User{}, err
+		return nil, err
 	}
 
 	err = repo.collection.FindOne(ctx, bson.M{"_id": uid}).Decode(&userEntity)
 	if err != nil {
+		if errors.As(err, &mongo.ErrNoDocuments) {
```

Comment:
We should use errors.Is instead of errors.As.

Reply:
errors.Is will check if the error is of a particular type, and error.As will check the chain of errors and try to extract it. So if if we have a chain of errors, the method Is will fail. But if have sure that is not the case I can change to use the method Is